category/category.go

```diff
@@ -0,0 +1,262 @@
+package category
+
+import (
+	"context"
+
+	"github.com/almighty/almighty-core/convert"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/log"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Defines "type" string to be used while validating jsonapi spec based payload
+const (
+	APIStringType = "categories"
+)
+
+// String constants for system categories
+const (
+	PlannerRequirements = "planner.requirements"
+	PlannerIssues       = "planner.issues"
+)
+
+// Do not change these UUIDs!!!
+// System defined categories
+var (
+	PlannerRequirementsID = uuid.FromStringOrNil("04aef834-1505-44cf-80e4-ab0d857d9f56") // "planner.requirements"
+	PlannerIssuesID       = uuid.FromStringOrNil("27d92fe4-b2ee-45c2-b9bb-01f355ad616f") // "planner.issues"
+)
+
+// Category describes a single category
+type Category struct {
+	gormsupport.Lifecycle
+	ID   uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	Name string
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *Category) TableName() string {
+	return "categories"
+}
+
+// WorkItemTypeCategoryRelationship describes relationship between a category and a workitemtype.
+type WorkItemTypeCategoryRelationship struct {
+	gormsupport.Lifecycle
+	ID             uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	CategoryID     uuid.UUID `sql:"type:uuid"`
+	WorkitemtypeID uuid.UUID `sql:"type:uuid"`
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *WorkItemTypeCategoryRelationship) TableName() string {
+	return "workitemtype_categories"
+}
+
+// Ensure fields implements the Equaler interface
+var _ convert.Equaler = Category{}
+var _ convert.Equaler = (*Category)(nil)
+
+// Equal returns true if two Category objects are equal; otherwise false is returned.
+func (category Category) Equal(u convert.Equaler) bool {
```

Comment:
@DhritiShikhar please your fixup commits smaller and dedicated to the concrete comment. Your commit includes more changes. Than that. Otherwise it is fine.